File: tyama_icpc2012dA.cs
using System;
using System.Linq;

class icpc2012dA{
	static void Main(){
		int i,n;
		n=int.Parse(Console.ReadLine());
		for(i=0;i<n;i++){
			int[] a=Console.ReadLine().Split(' ').Select(s=>int.Parse(s)).ToArray();
			a[0]-=1;a[1]-=1;
			Console.WriteLine(196471-a[0]*195-a[0]/3*5-a[1]*20+(a[0]%3!=2?a[1]/2:0)-a[2]);
		}
	}
}

File: tyama_icpc2012dA.ex
#!/usr/bin/env elixir
defmodule ICPC2012dA do
	def main2(n) do
		if n>0 do
			a=String.split(IO.chardata_to_string(IO.gets(''))," ")
			y=elem Integer.parse(Enum.at(a,0)),0
			m=elem Integer.parse(Enum.at(a,1)),0
			d=elem Integer.parse(Enum.at(a,2)),0
			IO.puts 196471-((y-1)*195)-div(y-1,3)*5-(m-1)*20+(if rem(y,3)==0 do 0 else div(m-1,2) end)-d
			main2(n-1)
		end
	end
	def main() do
		n=elem Integer.parse(IO.gets('')),0
		main2(n)
	end
end
ICPC2012dA.main